File: de.unistuttgart.iste.sqa.mpw.hamstersimulator.cpp/core/test/LocationTest.cpp
#include "gtest/gtest.h"

#include "HamsterCore.h"

#include <memory>
#include <unordered_set>

using namespace mpw;
using namespace hamster;

/// \note use 'NO-LINT' comment at tests to suppress a warning caused by TEST_F
class LocationTest : public testing::Test {
};

TEST_F(LocationTest, givenLocation0x0_whenCheckEqualsTo0x0_thenIsTrue) { /* NOLINT */
    auto sut = Location::from(0, 0);
    bool actual = sut == Location::from(0, 0);
    EXPECT_TRUE(actual);
}

TEST_F(LocationTest, givenLocation1x1_whenCheckEqualsTo2x1_thenIsFalse) { /* NOLINT */
    auto sut = Location::from(1, 1);
    bool actual = sut == Location::from(2, 1);
    EXPECT_FALSE(actual);
}

TEST_F(LocationTest, givenLocation3x5_whenPutIntoHashSet_andCheckIfSetContains3x5_thenCheckIsTrue) { /* NOLINT */
    std::unordered_set<Location> hashSet;
    hashSet.insert(Location::from(3, 5));
    bool actual = hashSet.find(Location::from(3, 5)) != hashSet.end();
    EXPECT_TRUE(actual);
}

TEST_F(LocationTest, givenLocation3x5_whenPutIntoHashSet_andCheckIfSetContains3x3_thenCheckIsFalse) { /* NOLINT */
    std::unordered_set<Location> hashSet;
    hashSet.insert(Location::from(3, 5));
    bool actual = hashSet.find(Location::from(3, 3)) != hashSet.end();
    EXPECT_FALSE(actual);
}

File: simulators/de.unistuttgart.iste.sqa.mpw.framework.java/core/src/test/java/de/unistuttgart/iste/sqa/mpw/framework/LocationTest.java
package de.unistuttgart.iste.sqa.mpw.framework;

import de.unistuttgart.iste.sqa.mpw.framework.mpw.Location;
import org.junit.jupiter.api.Test;

import java.util.HashSet;

import static org.junit.jupiter.api.Assertions.*;

public class LocationTest {
    @Test
    public void givenLocation0x0_whenCheckEqualsTo0x0_thenIsTrue() {
        var sut = Location.from(0, 0);
        boolean actual = sut.equals(Location.from(0, 0));
        assertTrue(actual);
    }

    @Test
    public void givenLocation1x1_whenCheckEqualsTo2x1_thenIsFalse() {
        var sut = Location.from(1, 1);
        boolean actual = sut.equals(Location.from(2, 1));
        assertFalse(actual);
    }

    @Test
    public void givenLocation3x5_whenPutIntoHashSet_andCheckIfSetContains3x5_thenCheckIsTrue() {
        var hashSet = new HashSet<Location>();
        hashSet.add(Location.from(3, 5));
        boolean actual = hashSet.contains(Location.from(3, 5));
        assertTrue(actual);
    }

    @Test
    public void givenLocation3x5_whenPutIntoHashSet_andCheckIfSetContains3x3_thenCheckIsFalse() {
        var hashSet = new HashSet<Location>();
        hashSet.add(Location.from(3, 5));
        boolean actual = hashSet.contains(Location.from(3, 3));
        assertFalse(actual);
    }
}